### DISCUSSION TITLE
Suspense not working in layout.tsx?

### DISCUSSION THREAD

**Discussion by whydea** [vercel/next.js#44385] [App Router]
Suspense not working in layout.tsx?

Example:
import { Orbitron } from '@next/font/google';
import { getConfigInstitution } from '../services/configInstitution';
import { loadConfigTheme } from '../utils/themeUtils';
import { Suspense } from 'react';

const orbitron = Orbitron({
   weight: ['400', '700'],
   style: ['normal']

async function getInitialConfig() {
   const configRes = await getConfigInstitution();

   const themeRes = loadConfigTheme(configRes);

   return {
     config: configRes,
     theme: themeRes

export default async function RootLayout({
   children,
   children: React.ReactNode
   const { config, theme } = await getInitialConfig()

   return(
     <html lang="en" className={orbitron.className}>
       <head />

       <body>
             <ThemeProvider config={config} theme={theme}>
               <Suspense fallback={<Loading />} >
                 {children}
              </Suspense>
             </ThemeProvider>
       </body>

     </html>

Using the same example on any page the suspense works, but not on Layout files.

I look for the initial settings in the layout file because I share it in a context so I don't have to look for them on each page.

The problem is not related to the ThemeProvider, as I've tried using Suspense only with {children} and it doesn't work either.

Is there any other strategy to show loading feedback to user on initial load?

**Comment by ciruz** [comment]
Hey,
checkout ``Templates`` ( https://beta.nextjs.org/docs/routing/pages-and-layouts#templates ), the docs say to ``Templates``:

> There may be cases where you need those specific behaviors, and templates would be a more suitable option than layouts. For example:
> - To change the default framework behavior. For example, Suspense Boundaries inside layouts only show the fallback the first time the Layout is loaded and not when switching pages. For templates, the fallback is shown on each navigation.

**Reply by whydea** [reply]
Hello @ciruz 
But apparently the suspense should work in the layout.tsx file.
I tested your suggestion of using suspend in the template file and it doesn't work either

**Reply by growupanand** [reply]
Hey @whydea , I am using next js app router, i am doing the same thing like i have a layout.tsx (server component) where i am fetching some data, in same folder i have loading.tsx, but loading.tsx content never shows.
Do you find any solution for this?

**Comment by FiorellaChilcon** [comment]
I have the same issue, Suspense is not working in `layout.tsx` files when making request within that file
